src/lib.rs

```diff
@@ -424,6 +424,41 @@ pub fn get_all_enclave_names() -> NitroCliResult<Vec<String>> {
     Ok(objects)
 }
 
+/// Sends the name to all the running enclaves and expects a response
+/// with the ID of the one that uniquely matched
+pub fn get_id_by_name(name: String) -> NitroCliResult<String> {
+    let (comms, _) = enclave_proc_command_send_all::<String>(
+        EnclaveProcessCommandType::GetIDbyName,
+        Some(&name),
+    )
+    .map_err(|e| {
+        e.add_subaction("Failed to send GetIDbyName command to all enclave processes".to_string())
+            .set_action("Get Enclave Names".to_string())
+    })?;
+
+    let mut replies: Vec<UnixStream> = vec![];
+    replies.extend(comms);
+    let objects: Vec<String> = enclave_proc_handle_outputs::<String>(&mut replies)
+        .iter()
+        .map(|v| v.0.clone())
+        .collect();
+
+    // Check if the name was not found or if there are multiple matches
+    if objects.is_empty() {
+        return Err(new_nitro_cli_failure!(
+            "No enclave matched the given name.".to_string(),
+            NitroCliErrorEnum::EnclaveNamingError
+        ));
+    } else if objects.len() > 1 {
+        return Err(new_nitro_cli_failure!(
+            "Conflicting enclave names have been found.".to_string(),
+            NitroCliErrorEnum::EnclaveNamingError
+        ));
+    }
```

Comment:
Maybe you could rewrite like this so there is no duplicating code
```suggestion
    if (objects.len() != 1) {
        return Err(new_nitro_cli_failure!(
            match objects.len() {
                0 => "No enclave matched the given name.".to_string(),
                2.. => "Conflicting enclave names have been found.".to_string(),
            },
            NitroCliErrorEnum::EnclaveNamingError
        ));
    }
```